### ISSUE TITLE
Mechanism for sites to indicate their preferred bridge

### ISSUE THREAD

**Issue by snarfed** [gugray/rss-parrot#29]
Hi again! Now that we have multiple feed-to-fediverse bridges - [RSS Parrot](https://rss-parrot.net/), [Bridgy Fed](https://fed.brid.gy/), [MastoFeed](https://mastofeed.org/), [rss-to-activitypub](https://github.com/dariusk/rss-to-activitypub), [feed2toot](https://feed2toot.readthedocs.io/), [feed2fedi](https://codeberg.org/MarvinsMastodonTools/feed2fedi) - we'll inevitably end up with multiple fediverse accounts for the same web site, each run by a different bridge. Not a big problem, but it'd still be nice to avoid.

One approach could be to let sites indicate their "preferred" bridge with eg `<link rel="me bridge">` in their home page's HTML, similar to #16. If example.com wants to use RSS Parrot to get its posts into the fediverse, it would add something like this in its HTML:

```html
<link rel="me bridge" href="https://rss-parrot.net/web/feeds/example.com">
```

Then, when other bridges start to create a fediverse account for example.com, they'd first check its home page HTML for a `link rel=bridge`. If one exists, and it doesn't point to them, they'd stop and give up.

One difficulty is that a site could want to use multiple bridges for different networks, eg RSS Parrot for the fediverse and https://atomstr.data.haus/ for Nostr. Maybe we'd allow multiple `<link rel="me bridge">`s to handle that, maybe differentiated by `type`, or maybe that differentiation isn't necessary.

Thoughts?

**Comment by gugray** [comment]
Thanks for these suggestions too! One part of me really likes the idea. The other part, which is a ruthless pragmatist, thinks that this would be largely wasted effort until there is some sort of popular awareness of the existence of these bridges. In reality I expect we would be getting very few, if any, such signals from feeds and websites out there.

Or to put it another way, this is a "social" task, not a technical one. For now I'm inclined to put it on the back burner...

**Comment by snarfed** [comment]
Agreed on all points. This specific proposal also requires web site owners to modify their sites' raw HTML `<head>`, which may not be easy for people on hosted services or high level servers like micro.blog, WordPress, Tumblr, Squarespace, Wix, etc.

On the plus side, re prioritization and effort, if we already implement opt out support, it hopefully won't be too much extra work to extract a specific bit of markup like this from a site's HTML and add them to our existing blocklists if necessary.

I'd also ideally like a version of this for bridged accounts that aren't web sites. For example, if multiple Bluesky <=> fediverse bridges emerge, it'd be nice for a Bluesky account to be able to indicate their preferred bridge too. They could put that in a link on their profile, but when bridges load their profile, they won't easily be able to distinguish "preferred bridge" links from other links. I'm not sure how to solve that yet.

**Comment by jg10-mastodon-social** [comment]
I've recently got my head around https://www.w3.org/TR/websub/ (previously PubSubHubbub). While it's positioned as an alternative to ActivityPub (https://www.w3.org/TR/social-web-protocols/#subscribing) it's interesting that it specifically implements the case where subscriptions are delegated to a separate hub, similar to the OP proposal that sites delegate to a preferred bridge.

The commercial service https://superfeedr.com provides both publisher- and subscriber- oriented "bridges". Publishers can use superfeedr as their nominated hub. Subscribers can get push notifications aggregated across all their subscriptions.
